File: Viewpager2/app/src/main/java/com/example/android/viewpager2/MainActivity.java
package com.example.android.viewpager2;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.viewpager2.widget.ViewPager2;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.view.ContextMenu;
import android.view.Menu;
import android.view.MenuItem;

import com.google.android.material.bottomnavigation.BottomNavigationView;
import com.google.android.material.navigation.NavigationBarView;

public class MainActivity extends AppCompatActivity {

    private Context mContext = MainActivity.this;
    private static final int ACTIVITY_NUM = 0;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        setupBottomNavigation();
        setupViewPager();

    }

    private void setupViewPager()
    {
        CustomPagerAdapter pagerAdapter = new CustomPagerAdapter(this);
        pagerAdapter.addFragment(new Fragment1());
        pagerAdapter.addFragment(new Fragment2());

        ViewPager2 viewPager2 = (ViewPager2) findViewById(R.id.main_viewpager2);
        viewPager2.setAdapter(pagerAdapter);
    }

    private void setupBottomNavigation()
    {
        BottomNavigationView bottomNavigationView = (BottomNavigationView) findViewById(R.id.bottom_navigation);
        BottomNaviagtionHelper.enableNavigation(mContext,bottomNavigationView);
        Menu menu = bottomNavigationView.getMenu();
        MenuItem menuItem = menu.getItem(ACTIVITY_NUM);
        menuItem.setChecked(true);
    }
}

File: File_IO_SharedPreferences/app/src/main/java/com/example/android/file_inout/MainActivity.kt
package com.example.android.file_inout

import android.content.Context
import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.Button
import com.google.android.material.snackbar.Snackbar
import java.io.File

class MainActivity : AppCompatActivity() {

    private val TAG : String = "File I/O Activity"

    private lateinit var file : File
    private lateinit var preferenceButton : Button

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        preferenceButton = findViewById(R.id.preference_button)

        preferenceButton.setOnClickListener {
            val intent = Intent(this, Main2Activity::class.java)
            startActivity(intent)
        }
    }


    private fun easyReadTextFile(fullPath: String): String {
        var contents = ""
        //context.openFileInput("").bufferedREader().useLines {lines ->
        //    content = lines.joinToString("\n")
        //}
        return contents
    }

    private fun easyWriteTextFile(filename : String, contents: String) {
        //context.openFileOutput(,Context.MODE_PRIVATE).use { stream ->
        //    stream.write(contents.toByteArray())
        //}
    }
}

